server/src/main/java/com/thoughtworks/go/server/service/UserService.java

```diff
@@ -573,4 +586,15 @@ private boolean validate(LocalizedOperationResult result, User user) {
         return false;
     }
 
+    private boolean hasUserChanged(User newUser, User originalUser) {
```

Comment:
Is an equals check not good enough?

Reply:
We only wanted to update the user in case `displayName` or `email` changed not `username`. Hence, the explicit checks.